src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UrlEncoder.cs

```diff
@@ -165,11 +165,11 @@ public unsafe override bool TryEncodeUnicodeScalar(int unicodeScalar, char* buff
             if (!WillEncode(unicodeScalar)) { return TryWriteScalarAsChar(unicodeScalar, buffer, bufferLength, out numberOfCharactersWritten); }
 
             numberOfCharactersWritten = 0;
-            uint asUtf8 = (uint)UnicodeHelpers.GetUtf8RepresentationForScalarValue((uint)unicodeScalar);
+            uint asUtf8 = unchecked((uint)UnicodeHelpers.GetUtf8RepresentationForScalarValue((uint)unicodeScalar));
```

Comment:
unchecked{} block here instead of statement (or add two unchecked statements) to include unicodeScalar